src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java

```diff
@@ -247,6 +252,36 @@ public void testExecuteOnWithNoDiffReport() throws Exception {
     verify(stashRequestFacade, times(0)).resetPullRequestApproval(eq(pr), (StashClient)Mockito.anyObject());
   }
 
+  @Test
+  public void testShouldApprovePullRequest() {
+    Issue minorIssue = new DefaultIssue().setSeverity(Severity.MINOR);
+    Issue majorIssue = new DefaultIssue().setSeverity(Severity.MAJOR);
+
+    report = new ArrayList<>();
+
+    report.add(minorIssue);
+    report.add(majorIssue);
+
+    assertFalse(
+        StashIssueReportingPostJob.shouldApprovePullRequest(Optional.empty(), report)
+    );
+
+    report.clear();
+    assertTrue(
+        StashIssueReportingPostJob.shouldApprovePullRequest(Optional.empty(), report)
+    );
+
+    report.add(minorIssue);
+    assertTrue(
+        StashIssueReportingPostJob.shouldApprovePullRequest(Optional.of(Severity.MINOR), report)

```

Comment:
The method is made a package-scoped static method especially to improve testing.
There is no need for any fixtures or the mocking of dependencies.